graph-api.yaml

```diff
@@ -0,0 +1,459 @@
+      type: http
+      scheme: bearer
+      bearerFormat: JWT
+    AwsBearerAuth:
+      description: |
+        For AWS usage we support JWTs that are injected into requests by the ELB.  
+        Since these come in a non-standard HTTP header we have to declare this as
+        an API Key scheme in OpenAPI spec as it doesn't have a notion of using a
+        HTTP scheme in a non-standard way (yay AWS weirdness!)
+      type: apiKey
+      in: header
+      name: X-Amzn-Oidc-Data
+  parameters:
+      dataset:
+        name: dataset
+        in: query
+        required: true
+        schema:
+          type: string
+        description: The underlying dataset (i.e. Graph) to apply requests against.
+      graphQLQuery:
+          name: query
+          in: query
+          required: true
+          schema:
+            type: string
+          description: The GraphQL query string.
+      sparqlQuery:
+          name: query
+          in: query
+          required: true
+          schema:
+            type: string
+          description: The SPARQL query string.
+      operationName:
+        name: operationName
+        in: query
+        required: false
+        schema:
+          type: string
+        description: Name of the operation to execute (optional).
+      variables:
+        name: variables
+        in: query
+        required: false
+        schema:
+          type: string
+          description: JSON-encoded variables for the query (optional).
+      extensions:
+        name: extensions
+        in: query
+        required: false
+        schema:
+          type: string
+        description: Extensions for the GraphQL request (optional).
+
+  requestBodies:
+      GraphQLRequest:
+        description: GraphQL request payload.
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                query:
+                  type: string
+                operationName:
+                  type: string
+                variables:
+                  type: object
+                extensions:
+                  type: object
+              required:
+                - query
+
+  responses:
+      GraphQLSuccess:
+        description: Successful response containing query results.
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                data:
+                  type: object
+      GraphQLError:
+        description: Error response for invalid GraphQL requests.
+        content:
+          application/problem+json:
+            schema:
+              type: object
+              properties:
+                type:
+                  type: string
+                title:
+                  type: string
+                status:
+                  type: integer
+                detail:
+                  type: string
+  schemas:
+    ValidationReport:
```

Comment:
Again need to check with @robwtelicent - I had it in my head that the validation report was itself an RDF file but I could be wrong??

Reply:
It can take a few of the RDF formats, I'll update it

Reply:
Yes, the validation report is Turtle RDF. The spec mentions other formats but AFAIK Fuseki only supports Turtle.

Reply:
Example:
```
PREFIX rdf:  <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX sh:   <http://www.w3.org/ns/shacl#>

[ rdf:type     sh:ValidationReport;
  sh:conforms  true
] .
```